sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TracezDataAggregator.java

```diff
@@ -0,0 +1,363 @@
+        filteredSpans.add(span.toSpanData());
+      }
+    }
+    return filteredSpans;
+  }
+
+  /**
+   * A class of boundaries for the latency buckets. The completed spans with a status of {@link
+   * io.opentelemetry.trace.Status#OK} are categorized into one of these buckets om the traceZ
+   * zPage.
+   */
+  enum LatencyBoundaries {
+    /** Stores finished successful requests of duration within the interval [0, 10us). */
+    ZERO_MICROSx10(0, TimeUnit.MICROSECONDS.toNanos(10)),
+
+    /** Stores finished successful requests of duration within the interval [10us, 100us). */
+    MICROSx10_MICROSx100(TimeUnit.MICROSECONDS.toNanos(10), TimeUnit.MICROSECONDS.toNanos(100)),
+
+    /** Stores finished successful requests of duration within the interval [100us, 1ms). */
+    MICROSx100_MILLIx1(TimeUnit.MICROSECONDS.toNanos(100), TimeUnit.MILLISECONDS.toNanos(1)),
+
+    /** Stores finished successful requests of duration within the interval [1ms, 10ms). */
+    MILLIx1_MILLIx10(TimeUnit.MILLISECONDS.toNanos(1), TimeUnit.MILLISECONDS.toNanos(10)),
+
+    /** Stores finished successful requests of duration within the interval [10ms, 100ms). */
+    MILLIx10_MILLIx100(TimeUnit.MILLISECONDS.toNanos(10), TimeUnit.MILLISECONDS.toNanos(100)),
+
+    /** Stores finished successful requests of duration within the interval [100ms, 1sec). */
+    MILLIx100_SECONDx1(TimeUnit.MILLISECONDS.toNanos(100), TimeUnit.SECONDS.toNanos(1)),
+
+    /** Stores finished successful requests of duration within the interval [1sec, 10sec). */
+    SECONDx1_SECONDx10(TimeUnit.SECONDS.toNanos(1), TimeUnit.SECONDS.toNanos(10)),
+
+    /** Stores finished successful requests of duration within the interval [10sec, 100sec). */
+    SECONDx10_SECONDx100(TimeUnit.SECONDS.toNanos(10), TimeUnit.SECONDS.toNanos(100)),
+
+    /** Stores finished successful requests of duration greater than or equal to 100sec. */
+    SECONDx100_MAX(TimeUnit.SECONDS.toNanos(100), Long.MAX_VALUE);
+
+    private final long latencyLowerBound;
+    private final long latencyUpperBound;
+
+    /**
+     * Constructs a {@code LatencyBoundaries} with the given boundaries and label.
+     *
+     * @param latencyLowerBound the latency lower bound of the bucket.
+     * @param latencyUpperBound the latency upper bound of the bucket.
+     */
+    LatencyBoundaries(long latencyLowerBound, long latencyUpperBound) {
+      this.latencyLowerBound = latencyLowerBound;
+      this.latencyUpperBound = latencyUpperBound;
+    }
+
+    /**
+     * Returns the latency lower bound of the bucket.
+     *
+     * @return the latency lower bound of the bucket.
+     */
+    long getLatencyLowerBound() {
+      return latencyLowerBound;
+    }
+
+    /**
+     * Returns the latency upper bound of the bucket.
+     *
+     * @return the latency upper bound of the bucket.
+     */
+    long getLatencyUpperBound() {
+      return latencyUpperBound;
+    }
+  }
+
+  static final class SpanBuckets {
```

Comment:
This file is getting really big. I'd pull the inner classes out to the top level (but keep them package access).

Reply:
This leads to a bunch of warnings
```
warning: auxiliary class SpanBuckets in /usr/local/google/home/wilhu/Documents/opentelemetry-java/sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TracezDataAggregator.java should not be accessed from outside its own source file
```
that cause the build to fail. Should I instead create a separate file for them?

Reply:
I've never seen an error like this before. How did you do the class move?

Reply:
I just moved the classes outside. They're on the same level as `TracezDataAggregator`.

Reply:
oh, they should definitely be in their own file. Sorry, I missed your question. Java allows having multiple top-level classes in the same file, but it is definitely strongly frowned upon.

Reply:
Which approach should I implement?
1. Leave the LatencyBoundaries class within TracezDataAggregator and pull the SpanBuckets class out
2. Pull both classes out and create a new file with the LatencyBoundaries class in SpanBuckets
3. Pull both classes out and create two new files (one for each class)

Reply:
I would definitely go with one file per class in this case. It's easier to understand (and test) smaller chunks of code when they aren't all in the same file.

Reply:
BTW, if you're using IDEA, you should just be able to use the `move` refactoring and have the IDE manage all the details for you.

Reply:
Alright, I separated the classes. I didn't write any new unit tests for `TracezSpanBuckets` because the two classes that call the functions are already fully tested.